spec/chewy/fields/base_spec.rb

```diff
@@ -402,33 +402,234 @@
+          end
+        end
+
+        let(:country_with_cities) do
+          cities = [
+            City.create!(id: 1, name: '', surname: '', description: nil)
+          ]
 
-        Country.create!(id: 1, cities: cities)
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1, 'description' => nil}
+            ]
+          )
+        end
       end
 
-      specify do
-        expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq('id' => 1, 'cities' => [
-          {'id' => 1, 'name' => 'City1'}, {'id' => 2, 'name' => 'City2'}
-        ])
+      context 'parental and children fields with ignore_blank flag' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
+              field :id
+              field :cities, ignore_blank: true do
+                field :id
+                field :name
+                field :surname, ignore_blank: true
+                field :description
+              end
+            end
+          end
+        end
+
+        let(:country_without_cities) do
+          cities = []
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        let(:country_with_cities) do
+          cities = [
+            City.create!(id: 1, name: '', surname: '', description: nil)
+          ]
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_without_cities))
+            .to eq('id' => 1)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1, 'name' => '', 'description' => nil}
+            ]
+          )
+        end
+      end
+
+      context 'parental field without ignore_blank: true flag' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
+              field :id
+              field :cities do
+                field :id
+                field :name
+                field :surname
+                field :description
+              end
+            end
+          end
+        end
+
+        let(:country_with_cities) do
+          cities = []
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities))
+            .to eq('id' => 1, 'cities' => [])
+        end
+      end
+
+      xcontext 'geo_point type with ignore_blank: true flag' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
+              field :id
+              field :cities do
+                field :id
+                field :name
+                field :location, type: :geo_point, ignore_blank: true do
```

Comment:
As I mentioned before, it would be nice to test this combination as well: `type: :geo_point, ignore_blank: false`

Reply:
Thank you!

I have added this spec and described its output above.